### PR TITLE
Project setup, fixing some funcitons, re-run of -a- group

### PR DESCRIPTION
**Pull request by steeleb**
Okay KW!

Starting back at the beginning of this to try to fix some oversights in the code from the first go-round. Basically, I was missing some of the ROSS CLP reservoirs in the actual RS pull, so I had to track down why. Most of it seemed to be an issue in the lat/longs of the water quality file. This PR looks bigger than it is because it tracks ~500 lines of a .csv file. :)

This group of targets creates a bunch of shapefiles (probably not that necessary, but maybe nice to have for easy mapping?)  in `a_locs_poly_setup/out/` that are untracked and writes one .csv file that is tracked here. You will need the symlink set up for this group of targets, LMK if you need an invite - it's pointing to the following fp on my onedrive: `NASA-Northern/data/`. The actual money target here is `a_collated_points`, which has all the NW station locations as well as all the POI (center points) for all lakes/ponds/reservoirs in CLP > 0.01 ha and center points of the NW lakes/reservoirs that happen to be just outside the CLP HUC12. 

A few notes:

The new code in `a_locs_poly_setup/src/get_POI_centers.R` is a copy from the better-written/more robust version from lakeSR repo.

I had to add some if/else statements in `a_locs_poly_setup/src/get_polygons.R` to get things to work nicely after destroying my pipeline... basically, I use this function to get waterbody polygons from both HUC4's and HUC8's and I needed some better error handling. 

I don't know that you necessarily *need* to run this, but feel free if you want to have this up to date on your end. Specific ask is to make sure the output file stored in `a_locs_poly_setup/out/` seems right to you and contains what I say it should. All the points are in WGS84 EPSG:4326.

---- 

Progress on #9

### CODE REVIEW

File: config.yaml
Comment (kathryn-willi) [inline config.yaml:1]: does this need to be `config.yml`?

File: b_site_RS_data_acquisition.R [hunk 1]
```diff
@@ -19,12 +19,12 @@ yaml_file <- "nw-poudre-historical-config.yml"
 
 # Set up python virtual environment ---------------------------------------
 
-tar_source("b_RS_data_acquisition/py/pySetup.R")
+tar_source("b_site_RS_data_acquisition/py/pySetup.R")
```
Comment (kathryn-willi) [inline b_site_RS_data_acquisition.R:22]: in run_targets.Rmd, line 66 needs to be rewritten to source("b_site_RS_data_acquisition/py/pySetup.R")

File: _targets.R [hunk 2]
```diff
@@ -10,16 +10,16 @@ tar_option_set(
 # source functions
 tar_source(files = c(
   "a_locs_poly_setup.R",
-  "b_RS_data_acquisition.R",
-  "c_historical_RS_data_collation.R",
-  "d_calculate_handoff_coefficients.R",
-  "e_apply_handoff_coefficients.R",
-  "f_separate_NW_CLP_data.R"
+  "b_Site_RS_data_acquisition.R",
```
Comment (kathryn-willi) [inline _targets.R:13]:
```suggestion
  "b_site_RS_data_acquisition.R",
```

File: a_locs_poly_setup/src/get_polygons.R [hunk 3]
```diff
@@ -8,36 +8,58 @@
 #' @param HUC a 4-digit or longer HUC value
```
Comment (kathryn-willi) [inline a_locs_poly_setup/src/get_polygons.R]:
```suggestion
#' @param HUC a 4-digit or 8-digit HUC value
```

File: _targets.R (hunk 2 above)
Comment (kathryn-willi) [reply _targets.R:13]: ...quickly stopped reviewing anything past a_locs_poly_setup!

**Review by kathryn-willi — APPROVED** [review]
Looks great to me! All the code ran without an issue. I also reviewed each target quite thoroughly, and each output was what I expected. My one quite general comment is that there seems to be a lot of testing of transformations that may be unnecessary but also - better safe than sorry !

File: config.yaml
Comment (steeleb) [reply config.yaml:1]: pah-tay-toh, pah-tah-toh by technicality, but to keep in line with the other yaml files, i will change this! Thank you!

File: _targets.R (hunk 2 above)
Comment (steeleb) [reply _targets.R:13]: :) yup, those will all change in the next PR!

**Comment by steeleb** [comment]
You're a rockstar, KW! Thank you for diving in on this. 

Yeah, there are a lot of testing bits, in part because of the number of different formats of data coming in... and a number of different HUC-types being pulled. This is definitely a story of maybe {targets} isn't the best for data munging per se... but given data getting updated at least annually, I guess it makes sense. 

I'm merging, and will have another PR coming in shortly for the -b- group!

File: b_site_RS_data_acquisition.R (hunk 1 above)
Comment (steeleb) [reply b_site_RS_data_acquisition.R:22]: I'll nab this in the next PR!

**Pull request merged**